fvm/src/kernel/default.rs

```diff
@@ -460,6 +460,21 @@ where
         })
     }
 
+    fn recover_public_key(
+        &mut self,
+        hash: &[u8; SIG_MESSAGE_HASH_SIZE],
```

Comment:
This is more "flexible", but a bit less convenient. Are there any use-cases where we wouldn't want to take the _message_ itself, and hash it internally?